components/tidb_query_normal_expr/src/builtin_time.rs

```diff
@@ -444,6 +444,37 @@ impl ScalarFunc {
         Ok(None)
     }
 
+    #[inline]
+    pub fn add_date_and_duration<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let date = try_opt!(self.children[0].eval_duration(ctx, row));
+        let duration = try_opt!(self.children[1].eval_duration(ctx, row));
+        let overflow = Error::overflow("DURATION", &format!("({} - {})", &date, &duration));
+        let res = date.checked_add(duration).ok_or(overflow)?;
+        Ok(Some(Cow::Owned(res.to_string().into_bytes())))
+    }
+
+    #[inline]
+    pub fn add_date_and_string<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let date = try_opt!(self.children[0].eval_duration(ctx, row));
```

Comment:
why `eval_duration` instead of `eval_time`?

Reply:
The original code in TiDB is like (https://github.com/pingcap/tidb/blob/master/expression/builtin_time.go#L5551)

```Golang
func (b *builtinAddDateAndStringSig) evalString(row chunk.Row) (string, bool, error) {
	arg0, isNull, err := b.args[0].EvalDuration(b.ctx, row)
	if isNull || err != nil {
		return "", isNull, err
	}
	s, isNull, err := b.args[1].EvalString(b.ctx, row)
	if isNull || err != nil {
		return "", isNull, err
	}
    // ...
}
```

Here `eval_duration` is used as a date.

Reply:
I see. Where's the test come from?